transport.go

```diff
@@ -197,7 +193,12 @@ func (t *transport) Dial(ctx context.Context, raddr ma.Multiaddr, p peer.ID) (tp
 		remotePubKey:    remotePubKey,
 		remotePeerID:    p,
 		remoteMultiaddr: remoteMultiaddr,
-	}, nil
+	}
+	if t.gater != nil && !(t.gater.InterceptAccept(conn) && t.gater.InterceptSecured(n.DirOutbound, p, conn)) {
```

Comment:
I don't think we call `InterceptAccept` on dial, just `InterceptSecured`. The user may register an `InterceptAddrDial` to prevent the dial to an address in the first place.

Reply:
Ok, I removed the `InterceptAccept`.